### PR TITLE
Closes #144  Assignment6

### LINKED ISSUE #144: Static Assets, Partials, and TLS
#### Configure the application to use TLS/SSL
- Generate your own TLS/SSL certificate                                                                    
- Make "certs" directory in ./lib.                                                               
- Place the certificates you generate into "certs" directory.          

#### Configure two server connections
 - One labelled ['web'] on port 8000 with no TLS configured.                                      
 - The other labelled ['web-tls'] on port 8001 will use TLS.                                      
      Configure this connection to use the ssl certificates you just generated                       
      and placed in ./lib/certs                                                                      
    * Example server certificate configuration at [npm newww ](https://github.com/npm/newww/blob/13e26e0eaf5c6699f068bb5418d932af76be4aef/server.js)

#### Use TLS everywhere
Configure the application to always use TLS [Transport Layer Security] (http://en.wikipedia.org/wiki/Transport_Layer_Security).  <br/> Every http request should be redirected to https. 
Two references: 
* [@hueniverse on the issue](https://github.com/hapijs/hapi/issues/778)
* [related StackOverflow post](http://stackoverflow.com/questions/28650829/hapijs-using-both-http-and-https-on-one-connection)

### Configure the home plugin.
  - Make a new point in the home plugin called: "login"
    * Make login route display view template named login.html.
  - Use partials to display the login form
    * make a directory inside of the ./views called partials                          
    * configure the home plugin to use the ./views/partials directory to load partials from. 
        * In ./views/partials/ directory make a file named form.html.
           * form should have two inputs one type="text" named "username" and the second type="password" named "password".
          * form should have one submission button with text value "Login".                        
        * Inside login.html template load the partial file named "form.html" 

#### Use Static Assets                                                                          
- Make directory in root of project called "assets".                                            
- Inside the assets directory, make three directories:                                           
  "styles" for css, 
  "scripts" for JS scripts,                                                                      
   "images" for project images.                                                                   
    * Place Hapi University logo into the images directory                                           
      and display it on the home page (home.html). 
      (image is in the repo ./images/logo.png)
    * Place a CSS file in the stylesheets directory (styles).                                        
      Name css file styles.css:
       * Make one simple style:                                                                      
         h3 { font-size: 24px; font-weight: bold; }                                                  
       * Use the stylesheet in home.html. Make a title inside h3 tags.
    * Make a script named login.js and put in ./assets/scripts   
      * Make this script print a console message to the browser when the form button is clicked.                                    
    * In the home plugin configure the application to serve these static assets.
    * Make a login link in home.html that links to the login route.

#### Exended Due Date is June 12, 2015
Original due date was June 6, 2015.

#### Next After This Assignment
* Make POST request from our form. 
* Validate the post request with Joi. 
* Use hapi-auth-cookie to start session.
* Successful login goes to restricted access page. 
* Error handling of login failure

### PR DESCRIPTION
**Pull request by zoe-1**
Put this together hoping it will move everyone forward with the assignment.
Seems like everyone is in a rut with no passing PRs submitted yet.

* configured TLS and enforce TLS on all routes
* configured two server connections: web, web-tls
* new point in home plugin using view template login.html
* form loaded from partials directory ./view/partials
* static assets:
    - loaded from:
        * ./assets/scripts/
        * ./assets/images/
        * ./assets/styles/
    - used in:
        * ./views/home.html
        * ./views/login.html
* 100% test coverage

### CODE REVIEW

**Comment by zoe-1** [comment]
* Wanted to put more thought into:
   - testing if web-tls loaded properly.
   - testing if TLS certs loaded properly. 
   - and more but ran out of time.

Code review is welcomed!
Would love to know your thoughts.

Come on community! Let's get some more PRs submitted!
This is distributed learning and everyone's participation is needed :-)

Cheers!

File: test/home.js [hunk 1]
```diff
@@ -39,10 +95,17 @@ describe('/home', function () {
...
+        labels: ['web']
+    },
+    {
+        host: 'localhost',
+        port: 0,
+        labels: ['web-tls'],
+        tls: Config.tls
+    }],
     plugins: {
         './home': {}
```
Comment (hussainanjar) [inline test/home.js]: You haven't specified which labels this plugin should use ?

Comment (zoe-1) [inline test/home.js]: Thanks.  Will fix it.

File: lib/home.js [hunk 2]
```diff
@@ -17,9 +17,39 @@ exports.register = function (server, options, next) {
...
+        path: '/{assetpath*}',
+        handler: {
+            directory: {
+                path: './assets/'
+            }
+        }
+    });
+
+
+    server.route({
```
Comment (hussainanjar) [inline lib/home.js]: Should we do this ? As this was not part of assignment.

Comment (zoe-1) [inline lib/home.js]: That was in the static assets part of the assignment.
* In the home plugin configure the application to serve these static assets.

Comment (hussainanjar) [inline lib/home.js]: Actually this comment is on the route which is redirecting / to /home :smile:

Comment (zoe-1) [inline lib/home.js]: O. misunderstood the question.  
Yes, that was not in the assignment.
I do not plan to merge my PR for this assignment so I took the liberty to 
add that in.  I was thinking about forwarding / to /home or another route as a convenience
for developing the home page.  This way can swap in and out different routes to be the home page. What do you think? Is it worth doing?  I am not sure if there would be much of a performance hit.

Comment (hussainanjar) [inline lib/home.js]: I don't think there will be a huge performance hit but if we follow best practices then / should be our home page rather then redirecting to another route.

On the other hand its learning about hapi over here so its a good example to see how you can add permanent redirects.

**Pull request closed without merging**